### DISCUSSION TITLE
official replacement question

### DISCUSSION THREAD

**Discussion by ClashSAN** [Klace/stable-diffusion-webui-images-browser#4] [General]
Hi @Klace, Hi @AlUlkesh,

There are at least two fork replacements. let's swap one of them to https://github.com/AUTOMATIC1111/stable-diffusion-webui/wiki/Extensions-index, with credits to original owner @yfszzx 
 
maybe your changes can be merged with each other's
just want some info before the forks diverge

**Comment by AlUlkesh** [comment]
Hi @ClashSAN, hi @Klace,

I suspect merging might not be that easy, I just made some rather big changes on my side. And I see Klace has done the same on his.

I don't know... Klace, are you planning on supporting this for a bigger audience?

I wouldn't mind doing it on my repo, but if Klace has strong feelings about it - I'm not in it for the fame :wink:

**Comment by Klace** [comment]
Good morning! I have no issues with merging and don't mind either of us taking over to be honest. I would have liked to just do some PRs with the original but as we all know the creator seems to have dropped it. 

I made this fork mostly for myself and then added some features some people on Discord were asking for. So, I'm game with whatever people want to do :)

**Comment by AlUlkesh** [comment]
Good to hear. Well, how about this then. Let me see how difficult it really is to merge the two forks. Perhaps we might have to decide on some things that have to go one or the other way.

Once that's done, I suppose we can throw dice, which one the new master repo should be :smiley:

**Reply by Klace** [reply]
Sure :) One thing that needs to be done for my repo that I'd like to do before pushing to the main extension list is change the caching files. Probably don't need two anymore (originally I only added it for Aesthetic scoring but changes to all EXIF later) and the directory shouldn't be in root webui.  

My manual ranking system is also not quite polished, it can technically be removed. If we keep it then the ranking.json should also  move like the EXIF cache.

**Comment by AlUlkesh** [comment]
Okay, I finished merging it up to your commit f93cc4e. When you get a chance can you take a look at it, to see if your enhancements are all there?

https://github.com/AlUlkesh/stable-diffusion-webui-images-browser/tree/KlaceMerge

I have moved the two files to the extension's basedir.

A couple of questions:

- The exif cache generation took on my smaller folder about 3 minutes (9K images). That's a lot of waiting, without seeing what's going on. I think we should put in some kind of indicator and/or even an option for the user to disable that functionality, if he doesn't need it.
- I'm apparently a bit slow on your ranking option... How is this supposed to work:
![image](https://user-images.githubusercontent.com/99896447/215901892-22539242-37ab-4a29-851f-799ea46d55cb.png)
What happens if you press a ranking button? Will all visible images get the ranking?
- What was the Next Image After Ranking checkbox supposed to do?

Anyway, nice to see what you have been working on :smiley:

**Comment by Klace** [comment]
1. Definitely need to add a message/loading indicator for the caching (now
you can see why I did the .json cache because imagine doing that for every
search and sort :P) Once the initial cache is created it's much much faster.

2. Clicking a ranking button on the left side will assign the ranking to an
image. On the right side it's a filter so you can see everything you marked
as 5 or 4 etc.

3. I wanted it so that if you assigned a ranking to an image and the box is
checked it would move to the next image automatically. This would make it
easy to rank a whole folder.

**Comment by Klace** [comment]
A feature I wanted to add was the ability for users to add more sorting
options to the drop-down themselves. Let's say new exif data is added to
files and the user wants to sort by it, they could add the parameter to the
sort list rather than needing the maintainer to update the extension.
Basically would allow sorting on any parameter.

**Comment by AlUlkesh** [comment]
Hi @Klace I fixed a couple more things in the merged branch and
- moved the warning box to the top, so now it doubles as a loading indicator
- included the aes_cache-fix
- added a logging option to the setting, so now the previos prints are only logged if this is set to warning. This can later be used to implement debugging output.

Anything else you think needs to go in there? Otherwise I think it's ready to become "official".
 
I could merge it with my main branch now and ask for it to be put into the webui's list. Or did you have second thoughts and would prefer yours to be the official one?

**Reply by Klace** [reply]
I have no issue if you'd like to be the official one :) Please go ahead and merge.

For the prints they were just debug for myself, I didn't expect the fork to take hold but then I started on the instruct-pix2pix stuff. They could be removed entirely.

**Comment by AlUlkesh** [comment]
@ClashSAN Ok, I think it's ready, can you please put it into webui's extension list?

https://github.com/AlUlkesh/stable-diffusion-webui-images-browser

And thanks for alerting us to the fact, that we were about to diverge the extension.
